docs/presentations/DB-write-speed/db-write-speed.slide

```diff
@@ -0,0 +1,501 @@
+           
+-------------------------------------------------------------------------------------------------------------------
+-----------
+ Delete on rule_hit  (cost=0.42..8.44 rows=1 width=6) (actual time=0.035..0.037 rows=0 loops=1)
+   ->  Index Scan using rule_hit_pkey on rule_hit  (cost=0.42..8.44 rows=1 width=6) (actual time=0.033..0.034 rows=
+0 loops=1)
+         Index Cond: (((cluster_id)::text = '94288c61-b7db-4d23-9f81-ee88d376970d'::text) AND (org_id = 4403030))
+ Planning Time: 0.132 ms
+ Execution Time: 0.079 ms
+(5 rows)
+```
+
+
+
+## UPSERT into rule_hit table (cont.)
+
+```
+EXPLAIN ANALYZE INSERT INTO rule_hit(org_id, cluster_id, rule_fqdn, error_key, template_data)
+                       VALUES (...);
+                                            QUERY PLAN                                            
+--------------------------------------------------------------------------------------------------
+ Insert on rule_hit  (cost=0.00..0.01 rows=1 width=132) (actual time=9.333..9.335 rows=0 loops=1)
+   ->  Result  (cost=0.00..0.01 rows=1 width=132) (actual time=0.003..0.004 rows=1 loops=1)
+ Planning Time: 0.064 ms
+ Execution Time: 9.363 ms
+(4 rows)
+```
+
+- performed several times (once per rule hit)!
+
+
+
+## Some scenarios
+
+- report with 0 rule hits (pretty common)
+
+```
+UPSERT INTO report table:         36 ms
+DELETE FROM recommendation table: 10 ms
+INSERT INTO recommendation table:  0 ms
+DELETE FROM rule_hit table:        0 ms
+INSERT INTO rule_hit table:        0 ms
+---------------------------------------
+                                  46 ms (21 wr/second)
+```
+
+
+
+## Some scenarios (cont.)
+
+- report with 3 rule hits (middle ground)
+
+```
+UPSERT INTO report table:         36 ms
+DELETE FROM recommendation table: 10 ms
+INSERT INTO recommendation table: 30 ms (3*9.983 ms)
+DELETE FROM rule_hit table:        0 ms
+INSERT INTO rule_hit table:       28 ms (3*9.363 ms)
+---------------------------------------
+                                 104 ms (9.6 wr/second)
+```
+
+
+
+## Some scenarios (cont.)
+
+- report with 10 rule hits (upper 'practical' limit)
+
+```
+UPSERT INTO report table:         36 ms
+DELETE FROM recommendation table: 10 ms
+INSERT INTO recommendation table: 99 ms (10*9.983 ms)
+DELETE FROM rule_hit table:        0 ms
+INSERT INTO rule_hit table:       93 ms (10*9.363 ms)
+---------------------------------------
+                                 238 ms (4.2 wr/second)
+```
+
+
+
+## Possible improvements
+
+- vacuuming
+- cleanup old records
+- proper index for `recommendation` table
+- (minor) get rid of `.report` postfix in all tables
+- make sure we use Amazon EBS SSD-based storage type
+- make sure that Notification DB does not affect Aggregator DB
+- `INSERT OR REPLACE INTO` (upsert) not needed, simple `INSERT` is enough
+- multiple INSERT statements into one statement (?)
```

Comment:
Do you mean:
```
INSERT INTO table_name (column_list)
VALUES
    (value_list_1),
    (value_list_2),
    ...
    (value_list_n);
```
or using multiple `INSERT`?

Reply:
the first one (but needs to be checked if help)

Reply:
perfect